File: src/main/com/stockcomp/controller/StockController.java
package com.stockcomp.controller;

import com.stockcomp.document.SymbolDocument;
import com.stockcomp.response.HistoricPriceResponse;
import com.stockcomp.response.RealTimePriceResponse;
import com.stockcomp.response.SymbolSearchResponse;
import com.stockcomp.service.StockService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/stock")
@CrossOrigin(origins = "http://localhost:8000", allowCredentials = "true")
public class StockController {

    private final StockService stockService;

    public StockController(StockService stockService) {
        this.stockService = stockService;
    }

    @GetMapping("/search-symbol")
    public ResponseEntity<List<SymbolSearchResponse>> searchSymbol(@RequestParam String symbol){
        var results = stockService.searchSymbol(symbol);

        return ResponseEntity.ok(results);
    }

    @GetMapping("/historic-prices")
    public ResponseEntity<List<HistoricPriceResponse>> getHistoricPrices(@RequestParam String symbol){
        var result = stockService.getHistoricPriceList(symbol);

        return ResponseEntity.ok(result);
    }

    @GetMapping("/symbol-suggestions")
    public ResponseEntity<List<SymbolDocument>> getSymbolSuggestions(@RequestParam String query){
        var results = stockService.getSymbolSuggestions(query);

        return ResponseEntity.ok(results);
    }

    @GetMapping("/real-time-price")
    public ResponseEntity<RealTimePriceResponse> getRealTimePrice(@RequestParam String symbol){
        var result = stockService.getRealTimePrice(symbol);

        return ResponseEntity.ok(result);
    }
}

File: src/main/com/stockcomp/controller/StockController.kt
package com.stockcomp.controller

import com.stockcomp.document.SymbolDocument
import com.stockcomp.response.HistoricPrice
import com.stockcomp.response.RealTimePrice
import com.stockcomp.response.SymbolSearch
import com.stockcomp.service.SymbolService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/stock")
@CrossOrigin(origins = ["http://localhost:8000"], allowCredentials = "true")
class StockController(private val symbolService: SymbolService) {

    @GetMapping("/search-symbol")
    fun searchSymbol(@RequestParam symbol: String): ResponseEntity<List<SymbolSearch>> {
        val results = symbolService.searchSymbol(symbol)

        return ResponseEntity.ok(results)
    }

    @GetMapping("/historic-prices")
    fun getHistoricPrices(@RequestParam symbol: String): ResponseEntity<List<HistoricPrice>> {
        val result = symbolService.getHistoricPriceList(symbol)

        return ResponseEntity.ok(result)
    }

    @GetMapping("/symbol-suggestions")
    fun getSymbolSuggestions(@RequestParam query: String): ResponseEntity<Collection<SymbolDocument>> {
        val results = symbolService.getSymbolSuggestions(query)

        return ResponseEntity.ok(results)
    }

    @GetMapping("/real-time-price")
    fun getRealTimePrice(@RequestParam symbol: String): ResponseEntity<RealTimePrice> {
        val result = symbolService.getRealTimePrice(symbol)

        return ResponseEntity.ok(result)
    }
}